### ISSUE TITLE
Strange runtime graph when combining steps and DAG

### ISSUE THREAD

**Issue by rileyjbauer** [argoproj/argo#925]
**Is this a BUG REPORT or FEATURE REQUEST?**:
Bug Report

**What happened**:
I ran a workflow with the YAML at the bottom [1], and the runtime graph looks wrong.
![image](https://user-images.githubusercontent.com/34456002/43414363-605b139e-93e7-11e8-9504-6293462f450d.png)

**What you expected to happen**:

A single tree showing the execution of the workflow.

**Anything else we need to know?**:

After running this workflow and looking through its runtime JSON, it seems like certain parent/child connections within the workflow YAML are represented only via `boundaryID`s which point from children to parents. For example, in the graph shown above, `flip-again` is floating by itself as its own tree, but in the runtime JSON it is `boundaryID` points to `condition-1-child` (also missing from the graph) which in turn points to `condition-1`. If these links were present in the graph, we would get what I would expect from reading the YAML.

**Environment**:
- Argo version:

argoproj/argoui:v2.1.0-beta2 
argoproj/workflow-controller:v2.1.0-beta2

```
- Kubernetes version :
clientVersion:
  buildDate: 2018-06-27T20:17:28Z
  compiler: gc
  gitCommit: 91e7b4fd31fcd3d5f436da26c980becec37ceefe
  gitTreeState: clean
  gitVersion: v1.11.0
  goVersion: go1.10.2
  major: "1"
  minor: "11"
  platform: linux/amd64
serverVersion:
  buildDate: 2018-05-31T18:32:23Z
  compiler: gc
  gitCommit: 9b5b719c5f295c99de68ffb5b63101b0e0175376
  gitTreeState: clean
  gitVersion: v1.9.7-gke.3
  goVersion: go1.9.3b4
  major: "1"
  minor: 9+
  platform: linux/amd64
```

[1]

```
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: myflipcoin-
spec:
  arguments:
    parameters: []
  entrypoint: myflipcoin
  templates:
  - name: condition-1
    inputs:
      parameters:
      - name: flip-output
    steps:
    - - name: condition-1-child
        arguments:
          parameters:
          - name: flip-output
            value: '{{inputs.parameters.flip-output}}'
        template: condition-1-child
        when: '{{inputs.parameters.flip-output}} == heads'
  - name: condition-1-child
    dag:
      tasks:
      - name: condition-2
        arguments:
          parameters:
          - name: flip-again-output
            value: '{{tasks.flip-again.outputs.parameters.flip-again-output}}'
          - name: flip-output
            value: '{{inputs.parameters.flip-output}}'
        dependencies:
        - flip-again
        template: condition-2
      - name: flip-again
        arguments:
          parameters:
          - name: flip-output
            value: '{{inputs.parameters.flip-output}}'
        template: flip-again
    inputs:
      parameters:
      - name: flip-output
  - name: condition-2
    inputs:
      parameters:
      - name: flip-again-output
      - name: flip-output
    steps:
    - - name: condition-2-child
        arguments:
          parameters:
          - name: flip-again-output
            value: '{{inputs.parameters.flip-again-output}}'
          - name: flip-output
            value: '{{inputs.parameters.flip-output}}'
        template: condition-2-child
        when: '{{inputs.parameters.flip-again-output}} == tails'
  - name: condition-2-child
    dag:
      tasks:
      - name: print-tail
        arguments:
          parameters:
          - name: flip-again-output
            value: '{{inputs.parameters.flip-again-output}}'
          - name: flip-output
            value: '{{inputs.parameters.flip-output}}'
        template: print-tail
    inputs:
      parameters:
      - name: flip-again-output
      - name: flip-output
  - name: condition-3
    inputs:
      parameters:
      - name: flip-output
    steps:
    - - name: condition-3-child
        arguments:
          parameters:
          - name: flip-output
            value: '{{inputs.parameters.flip-output}}'
        template: condition-3-child
        when: '{{inputs.parameters.flip-output}} == tails'
  - name: condition-3-child
    dag:
      tasks:
      - name: print-tail1
        arguments:
          parameters:
          - name: flip-output
            value: '{{inputs.parameters.flip-output}}'
        template: print-tail1
    inputs:
      parameters:
      - name: flip-output
  - name: flip
    container:
      args:
      - python -c "import random; result = 'heads' if random.randint(0,1) == 0 else
        'tails'; print(result)" | tee /tmp/output
      command:
      - sh
      - -c
      image: python:alpine3.6
    outputs:
      parameters:
      - name: flip-output
        valueFrom:
          path: /tmp/output
  - name: flip-again
    container:
      args:
      - python -c "import random; result = 'heads' if random.randint(0,1) == 0 else
        'tails'; print(result)" | tee /tmp/output
      command:
      - sh
      - -c
      image: python:alpine3.6
    outputs:
      parameters:
      - name: flip-again-output
        valueFrom:
          path: /tmp/output
  - name: myflipcoin
    dag:
      tasks:
      - name: condition-1
        arguments:
          parameters:
          - name: flip-output
            value: '{{tasks.flip.outputs.parameters.flip-output}}'
        dependencies:
        - flip
        template: condition-1
      - name: condition-3
        arguments:
          parameters:
          - name: flip-output
            value: '{{tasks.flip.outputs.parameters.flip-output}}'
        dependencies:
        - flip
        template: condition-3
      - name: flip
        template: flip
  - name: print-tail
    container:
      command:
      - echo
      - '"it was tail"'
      image: alpine:3.6
  - name: print-tail1
    container:
      command:
      - echo
      - '"it was tail"'
      image: alpine:3.6

**Comment by jessesuen** [comment]
I think this may have been fixed in v2.1.1. I ran the workflow in the bug, and here are example outputs, which do not show disconnected graphs:

![image](https://user-images.githubusercontent.com/12677113/43554977-668cf4e6-95ac-11e8-86c3-deed6fb8b65f.png)

![image](https://user-images.githubusercontent.com/12677113/43555042-ca231058-95ac-11e8-88ca-4717934eb423.png)

**Comment by jessesuen** [comment]
Closing since I'm fairly certain this was resolved as part of the metadata fixes that went into v2.1.1.
